src/tests/Test/Compile.hs

```diff
@@ -133,4 +135,20 @@ case_charEnum = do
 defConf :: IO Config
 defConf = do
   cfg <- defaultConfigWithSandbox
+#if TYPESCRIPT
```

Comment:
Would be nice to move the CPP related to JS/TS into its own file e.g. have an `addMoreCompilerOptions` (with a better name) that toggles typescript